Added option for showing playing status on tray icon (initial sketches, need help)

TotalCaesar659 opened the pull request:
When I saw https://github.com/Alexey-Yakovenko/deadbeef/issues/1682 I've got an idea to create icons for playing, pause and stop for tray.
Examples:
![1](https://user-images.githubusercontent.com/14265316/29229989-b05fe4c8-7ee9-11e7-99ca-4f30451aec42.png)
![2](https://user-images.githubusercontent.com/14265316/29229993-b3b6fc56-7ee9-11e7-837a-fa1975150780.png)
![3](https://user-images.githubusercontent.com/14265316/29230000-b66f58da-7ee9-11e7-872b-53d01a55029c.png)
This PR is initial, tray icons change only after starting DeadBeeF according to saved status. I don't know how to change them when playing a track, so it needs help.
And I've added only 32x32 icons for checking will they suite users and getting comments. If they do, I'm ready to create and add another sizes.

Alexey-Yakovenko commented on plugins/gtkui/gtkui.c:
```diff
@@ -489,24 +489,47 @@ gtkui_update_status_icon (gpointer unused) {
-        trayicon = gtk_status_icon_new_from_file(iconpath);
+    int show_playing_status_on_tray_icon = deadbeef->conf_get_int ("gtkui.show_playing_status_on_tray_icon", 0);
+    
+    if (deadbeef->conf_get_int ("gtkui.show_playing_status_on_tray_icon", 1)) {
```
There are a few problems with this code. This function is called when configuration changes, but you need to change the icon when playback state changes.

Another problem is that previous icon object is never freed, so every time this code is called -- there would be a memory leak. However, this problem is present in the original code as well.

And last but not least, you don't want to read the files every time when playback state changes. Instead, you need to preload all icons at initialization time.

TotalCaesar659 replied:
Well, I'm not software engineer, so I did it for myself first. So, if somebody checks it and finishes my job - it would be great.

Alexey-Yakovenko commented:
The problem with the icons:
1. They don't look visually appealing enough to get accepted.
2. You provided only 32x32 icons, but all resolutions are necessary.
3. You need to add the new icons to icons/Makefile.am, so that they get installed.

TotalCaesar659 commented:
>They don't look visually appealing enough to get accepted.

I've created the icons based on original DeaDBeeF logo. In fact, I've re-drawn original SVG file and converted it to PNG.
![1](https://raw.githubusercontent.com/Alexey-Yakovenko/deadbeef/master/icons/32x32/deadbeef.png)
So they can't be less "visually appealing" than the original logo.

>You provided only 32x32 icons, but all resolutions are necessary.
>You need to add the new icons to icons/Makefile.am, so that they get installed.

I thought about it, but as I said, "I've added only 32x32 icons for checking will they suite users and getting comments. If they do, I'm ready to create and add another sizes."
So tell me, why they don't suit you. It would be good to get more concrete details (color, shadows etc.), not just "they don't look visually appealing enough". Many men, many minds.
Again, I already said about the original logo. Maybe, it should be re-drawn too with them.

Alexey-Yakovenko commented:
> Well, I'm not software engineer, so I did it for myself first. So, if somebody checks it and finishes my job - it would be great.

ok.. sorry, but there's nobody to finish your job.

Alexey-Yakovenko commented:
It's nothing personal, I just think that the icons need some extra work to make them look better. Right now the green cores look too blurry and oversized. Try making them more sharp and slightly smaller. I don't know how much it would help though -- not a designer myself.

Alexey-Yakovenko commented:
The reason why they're blurry is because of the glow that you carried from the original app icon.
You should try disabling this glow -- it doesn't work well with the kind of icons you're making, especially at this pixel size.
